lib/message_store/projection.ex

```diff
@@ -19,4 +21,13 @@ defmodule MessageStore.Projection do
       def handle_message(_, data), do: data
     end
   end
+
+  @spec load(String.t(), module(), fetch(), (any() -> any())) :: any()
+  def load(stream, projection, fetch, fun)
+      when is_binary(stream) and is_atom(projection) and is_function(fetch, 2) and
+             is_function(fun, 1) do
+    {:ok, data} = fetch.(stream, projection)
+
+    fun.(data)
```

Comment:
Not sure if it is correct. In case that `fetch` returns `{:error, any()}` then `load` function will failed. Maybe also the `load` should return `Result` type (or `Maybe`?)

Reply:
Yes, it is correct. This is required here. If the data cannot be read, the system is in an incorrect state and must be switched off as soon as possible. There is no need to further propagate the error.